packages/trpc/server/middlewares/sessionMiddleware.ts

```diff
@@ -102,21 +102,16 @@ export async function getUserFromSession(ctx: TRPCContextInner, session: Maybe<S
   // This helps to prevent reaching the 4MB payload limit by avoiding base64 and instead passing the avatar url
 
   const locale = user?.locale ?? ctx.locale;
-
-  const isOrgAdmin = !!user.profile?.organization?.members.filter(
-    (member) => (member.role === "ADMIN" || member.role === "OWNER") && member.userId === user.id
-  ).length;
+  const isOrgAdmin = await UserRepository.isAdminOrOwnerOfTeam({
+    userId: user.id,
+    teamId: user.profile.organization?.id ?? -1,
+  });
```

Comment:
Could we return the memberships that are related to the user? That way we can run this `isOrgAdmin` check through just those memberships instead of making a second DB query? Might solve your problem down below as well.

Reply:
That is the approach we're replacing tho. What other problem are you referring to? We really don't need tho send these memberships to the client. And we're only using them to check for the user role.

Reply:
We opted to merge and not overthink this @joeauyeung - this entire logic needs evaluation and a simple call like this won't have much effect.